File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/Abu_HadHoud_Solutions_Using_C++/Problem8.cpp
#include <bits/stdc++.h>
using namespace std;
enum enPassFail { Pass = 1, Fail = 2 };

int ReadMark() {
	int Mark;
	cout << "Please enter your Mark : ";
	cin >> Mark;
	return Mark;
}

enPassFail CheckMark(int Mark) {
	if (Mark >= 50)
		return enPassFail::Pass;
	else
		return enPassFail::Fail;

}
void PrintResult(int Mark) {
	if (CheckMark(Mark) == enPassFail::Pass)
		cout << "\nYou Passed";
	else
		cout << "\nYou Failed";

}

int main() {
	PrintResult(ReadMark());
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem8.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        enum enPassFail { Pass =1,Fail= 2 }
        static int ReadMark()
        {

            Console.Write("Please enter your Mark : ");
            int Mark = int.Parse(Console.ReadLine());

            return Mark;
        }
        static string CheckMark(int Mark)
        {
            if(Mark >= 50)
                return "Passed";
            else
                return "Failed";
        }
        static void PrintResult(int Mark)
        {
            Console.WriteLine($"\nYour Mark is {Mark} and you are {CheckMark(Mark)}");
        }
        static void Main(string[] args)
        {
            PrintResult(ReadMark());
        }
    }
}